skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,337 @@
+        self.score = score
+        self.base = base
+        self.natural_gradient = natural_gradient
+        self.n_estimators = n_estimators
+        self.learning_rate = learning_rate
+        self.minibatch_frac = minibatch_frac
+        self.col_sample = col_sample
+        self.verbose = verbose
+        self.verbose_eval = verbose_eval
+        self.init_params = None
+        self.n_features = None
+        self.base_models = []
+        self.scalings = []
+        self.col_idxs = []
+        self.tol = tol
+        self.random_state = check_random_state(random_state)
+        self.best_val_loss_itr = None
+        self.validation_fraction = validation_fraction
+        self.early_stopping_rounds = early_stopping_rounds
+
+        super().__init__()
+
+    def dist_to_ngboost_instance(self):
+        """
+        Convert string to NGBoost object.
+
+        self.dist the input string for the type of Distribution.
+        It then creates an object of that particular NGBoost Distribution.
+
+        Returns
+        -------
+        NGBoost Distribution object.
+        """
+        from ngboost.distns import Laplace, LogNormal, Normal, Poisson, T
+
+        dist_ngboost = Normal
+        if self.dist == "Normal":
+            dist_ngboost = Normal
+        elif self.dist == "Laplace":
+            dist_ngboost = Laplace
+        elif self.dist == "TDistribution":
+            dist_ngboost = T
+        elif self.dist == "Poisson":
+            dist_ngboost = Poisson
+        elif self.dist == "LogNormal":
+            dist_ngboost = LogNormal
+
+        return dist_ngboost
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        from ngboost import NGBRegressor
+        from ngboost.learners import default_tree_learner
+
+        dist_ngboost = self.dist_to_ngboost_instance()
+
+        self.ngb = NGBRegressor(
+            Dist=dist_ngboost,
+            Score=self.score,
+            Base=default_tree_learner,
+            natural_gradient=True,
+            n_estimators=self.n_estimators,
+            learning_rate=self.learning_rate,
+            minibatch_frac=self.minibatch_frac,
+            col_sample=self.col_sample,
+            verbose=self.verbose,
+            verbose_eval=self.verbose_eval,
+            tol=self.tol,
+            random_state=self.random_state,
+            validation_fraction=self.validation_fraction,
+            early_stopping_rounds=self.early_stopping_rounds,
+        )
+        self.ngb.fit(X, y)
+        self._is_fitted = True
+        return self
+
+    def _predict(self, X):
```

Comment:
I would recommend, remove `_predict` and rely on the default behaviour entirely.

Reply:
I think it would be useful to keep `_predict` in case for just predicting the values. I have not used this anymore in `_predict_proba`. Please do let me know if you think otherwise.

Reply:
hm, I suppose it makes sense - not a blocker from my side.